### PR TITLE
Here is the new MongoEngineStorage class

### LINKED ISSUE #34: Adding support for mongo_engine.MongoEngineStorage
I'm volunteering to write, test, & document this expansion, but wanted to raise it as a issue first to make sure of no complications.

Essentially, it is not too different than the `MongoPiggybackStorage` variant already written. But it would be referenced the manner someone using the _MongoEngine_ wrapper library would expect. Something like:

```
from flask.ext.track_usage import TrackUsage
from flask.ext.track_usage.storage.mongo_engine import MongoEngineStorage

t = TrackUsage(app, MongoEngineStorage())
```

The only reason MongoEngineStorage would be stored in `.storage.mongo_engine` rather than `.storage.mongo` would be to prevent a needless library dependency for those not using MongoEngine.

If the user were to want to change a characteristic of the storage they would do it with a MongoEngine class inheritance of the default. For example to change the name of the collection:

```
from flask.ext.track_usage import TrackUsage
from flask.ext.track_usage.storage.mongo_engine import MongoEngineStorage, UsageTracker

class MyTrackUsage(UsageTracker):
    meta = {'collection': 'websiteDUsage'}

t = TrackUsage(app, MongoEngineStorage(doc=MyTrackUsage))
```

I also have a second expansion idea, but I'm not sure it is best placed into this library. I'll post a second issue to discuss that.

### PR DESCRIPTION
**Pull request by JohnAD**
added Storage class; docs; testing

Next, I'll start on the summation functions. I'll probably begin with documentation and keep the scope limited.

Resolves #34

### CODE REVIEW

**Comment by ashcrow** [comment]
Will review soon. Apologies, I'm in the midst of traveling.

File: src/flask_track_usage/storage/mongo.py [hunk 1]
```diff
@@ -134,3 +134,133 @@ def set_up(
...
+
+        :Parameters:
+           - `doc`: optional alternate MongoEngine document class.
+           - 'website': name for the website. Defaults to 'default'. Useful
+             when multiple websites are saving data to the same collection.
+           - 'apache_log': if set to True, then an attribute called
+             'apache_combined_log' is set that mimics a line from a traditional
+             apache webserver web log text file.
+
+        .. versionchanged:: 2.0.0
```
Comment (ashcrow) [inline src/flask_track_usage/storage/mongo.py]: 👍 thank you for the ``versionchanged``

**Comment by JohnAD** [comment]
I suspect that Travis CI has to be told to include the mongoengine library in the config.

```
{
  "language": "python",
  "python": "2.7",
  "services": [
    "mongodb",
    "mongoengine"
  ],
```
...

**Comment by ashcrow** [comment]
> I suspect that Travis CI has to be told to include the mongoengine library in the config.

I believe you are right. Line 9 of ``.travis.yml``

**Comment by ashcrow** [comment]
LGTM

**Comment by JohnAD** [comment]
yes.

**Pull request merged**